src/main/java/net/minecraftforge/common/data/ForgeBiomeTagsProvider.java

```diff
@@ -9,104 +9,259 @@
-        tag(Tags.Biomes.IS_MOUNTAIN).addTag(Tags.Biomes.IS_PEAK).addTag(Tags.Biomes.IS_SLOPE);
-        tag(Tags.Biomes.IS_UNDERGROUND).addTag(Tags.Biomes.IS_CAVE);
+    protected void addTags(HolderLookup.Provider lookupProvider) {
+        tag(NO_DEFAULT_MONSTERS).add(Biomes.MUSHROOM_FIELDS).add(Biomes.DEEP_DARK);
+        tag(HIDDEN_FROM_LOCATOR_SELECTION); // Create tag file for visibility
+
+        tag(IS_VOID).add(Biomes.THE_VOID);
+
+        tag(IS_END).addTags(BiomeTags.IS_END);
+        tag(IS_NETHER).addTags(BiomeTags.IS_NETHER);
+        tag(IS_OVERWORLD).addTags(BiomeTags.IS_OVERWORLD);
+
+        tag(IS_HOT_OVERWORLD)
+                .add(Biomes.SWAMP)
+                .add(Biomes.MANGROVE_SWAMP)
+                .add(Biomes.JUNGLE)
+                .add(Biomes.BAMBOO_JUNGLE)
+                .add(Biomes.SPARSE_JUNGLE)
+                .add(Biomes.DESERT)
+                .add(Biomes.BADLANDS)
+                .add(Biomes.WOODED_BADLANDS)
+                .add(Biomes.ERODED_BADLANDS)
+                .add(Biomes.SAVANNA)
+                .add(Biomes.SAVANNA_PLATEAU)
+                .add(Biomes.WINDSWEPT_SAVANNA)
+                .add(Biomes.STONY_PEAKS)
+                .add(Biomes.WARM_OCEAN);
+        tag(IS_HOT_NETHER)
+                .add(Biomes.NETHER_WASTES)
+                .add(Biomes.CRIMSON_FOREST)
+                .add(Biomes.WARPED_FOREST)
+                .add(Biomes.SOUL_SAND_VALLEY)
+                .add(Biomes.BASALT_DELTAS);
+        tag(IS_HOT_END);
+        tag(IS_HOT).addTag(IS_HOT_OVERWORLD).addTag(IS_HOT_NETHER).addOptionalTag(IS_HOT_END.location());
+
+        tag(IS_COLD_OVERWORLD)
+                .add(Biomes.TAIGA)
+                .add(Biomes.OLD_GROWTH_PINE_TAIGA)
+                .add(Biomes.OLD_GROWTH_SPRUCE_TAIGA)
+                .add(Biomes.WINDSWEPT_HILLS)
+                .add(Biomes.WINDSWEPT_GRAVELLY_HILLS)
+                .add(Biomes.WINDSWEPT_FOREST)
+                .add(Biomes.SNOWY_PLAINS)
+                .add(Biomes.ICE_SPIKES)
+                .add(Biomes.GROVE)
+                .add(Biomes.SNOWY_SLOPES)
+                .add(Biomes.JAGGED_PEAKS)
+                .add(Biomes.FROZEN_PEAKS)
+                .add(Biomes.STONY_SHORE)
+                .add(Biomes.SNOWY_BEACH)
+                .add(Biomes.SNOWY_TAIGA)
+                .add(Biomes.FROZEN_RIVER)
+                .add(Biomes.COLD_OCEAN)
+                .add(Biomes.FROZEN_OCEAN)
+                .add(Biomes.DEEP_COLD_OCEAN)
+                .add(Biomes.DEEP_FROZEN_OCEAN);
+        tag(IS_COLD_NETHER);
+        tag(IS_COLD_END)
+                .add(Biomes.THE_END)
+                .add(Biomes.SMALL_END_ISLANDS)
+                .add(Biomes.END_MIDLANDS)
+                .add(Biomes.END_HIGHLANDS)
+                .add(Biomes.END_BARRENS);
+        tag(IS_COLD).addTag(IS_COLD_OVERWORLD).addOptionalTag(IS_COLD_NETHER.location()).addTag(IS_COLD_END);
```

Comment:
The biomes marked in tag `c:is_cold/end` are not marked as `c:is_cold` in Fabric. However, I think this is more of an oversight on Fabric's part than it is an actual discrepancy on your part. If you want to continue to have 100% parity in terms of only having entries that are true in both NeoForge and Fabric, the easiest way would be to change this line to simply this:

```java
tag(IS_COLD).addTag(IS_COLD_OVERWORLD);
```